curl/src/main/scala/org/http4s/curl/http/CurlRequest.scala

```diff
@@ -77,7 +78,10 @@ private[curl] object CurlRequest {
 
         ec.addHandle(handle.curl, recv.onTerminated)
       }
-    )
+    ) *>
+      // IMPORTANT NOTE: we need to keep a reference to send and receive until everything is finished
+      // As curl needs those objects to be alive and not removed by GC
+      GCRoot(send, recv)
```

Comment:
There's some tricky sequencing going on here. For example the `Resource[Response]` could be in the process of finalizing, and the `GCRoot` could be cleaned up. But then there is an auto-`cede`, and cURL gets more data on this handle, and tries to run the callback, but the data has been GCed.

Basically we need to be 100% confident that these pointers won't be dereferenced anymore once they are released from the GC.

Reply:
I moved the `GCRoot` resource before curl handler, so it is not released before curl handler is cleaned up, right? 
Does it solve this problem?

Reply:
Ah, clever! Yes that looks good to me.

Reply:
So I updated the PR description to reflect this change for documentation purposes.